City-of-Helsinki/helsinki-paatokset
# PP-249: Add default texts config form with banner texts.

**Testing backend functionality:**
- BEFORE checking out branch: run `make ahjo-migrations`
- Checkout branch, then run `make drush-cr drush cim`
- Check that the settings form exists, and that the values can be edited and translated: https://helsinki-paatokset.docker.so/fi/admin/tools/default-texts
- Edit some policymaker nodes and confirm they have content in the description fields: https://helsinki-paatokset.docker.so/fi/admin/content/decisionmakers
- SSH into the container and run: `drush ahjo-proxy:remove-policymaker-fields -v`
- Reload the policymaker nodes. The description fields should now be empty.

**Testing frontend functionality:**

Go to some policymaker page i.e.https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginhallitus

There should be the default texts visible before calendar, document, decisions and recording.
Edit the node and add some text to one of the section fields and check that the change appears on the page.

Check some case or decision node i.ex. https://helsinki-paatokset.docker.so/fi/asia/hel-2021-014994
The banner should have the content from the default content here https://helsinki-paatokset.docker.so/fi/admin/tools/default-texts

Edit the content and check that the change appears in the banner.
